iot_verifier/src/metrics.rs

```diff
@@ -0,0 +1,81 @@
+use std::cell::RefCell;
+
+const LOADER_BEACON_COUNTER: &str = "oracles_iot_verifier_loader_beacon";
+const LOADER_WITNESS_COUNTER: &str = "oracles_iot_verifier_loader_witness";
+const INVALID_WITNESS_COUNTER: &str = "iot_verifier_invalid_witness_report";
+const BEACON_GUAGE: &str = "oracles_iot_verifier_num_beacons";
+
+pub struct Metrics;
+
+impl Metrics {
+    pub fn count_loader_beacons(count: u64) {
+        metrics::counter!(LOADER_BEACON_COUNTER, count);
+    }
+
+    pub fn count_loader_witnesses(count: u64) {
+        metrics::counter!(LOADER_WITNESS_COUNTER, count);
+    }
+
+    pub fn count_loader_invalid_witnesses(count: u64, labels: &[(&'static str, &'static str)]) {
+        metrics::counter!(INVALID_WITNESS_COUNTER, count, labels);
+    }
+
+    pub fn num_beacons(count: f64) {
+        metrics::gauge!(BEACON_GUAGE, count);
```

Comment:
Maybe hide the f64 requirement for gauges by moving the casting into this function? I think it’s weird enough that a gauge requires a float but now that we’re obscuring the gauge behind our own function here it looks even more arbitrary that youre tracking a gauge with a float

Reply:
Good idea, fixed 857006f